nipype/interfaces/base/specs.py

```diff
@@ -314,6 +315,39 @@ def __all__(self):
         return self.copyable_trait_names()
 
 
+def _deepcopypatch(self, memo):
+    """
+    Replace the ``__deepcopy__`` member with a traits-friendly implementation.
+
+    A bug in ``__deepcopy__`` for ``HasTraits`` results in weird cloning behaviors.
+    Occurs for all specs in Python<3 and only for DynamicTraitedSpec in Python>2.
```

Comment:
These both mean Python 2.7...

Reply:
Not really - for Python 2.7 now this bug arises also with static specs. For Python 3 it keeps happening only with `DynamicTraitedSpec`

Reply:
Ah, you're right. I guess I was reading too quickly.